ktor-server/ktor-server-plugins/ktor-server-websockets/jvm/test/io/ktor/tests/websocket/WebSocketTest.kt

```diff
@@ -118,593 +106,402 @@ class WebSocketTest {
-                call.response.awaitWebSocket(Duration.ofSeconds(10))
+                send(bytes)
             }
         }
-    }
-
-    @Test
-    fun testMasking() {
-        withTestApplication {
-            application.install(WebSockets)
-            application.routing {
-                webSocketRaw("/echo") {
-                    masking = true
-
-                    incoming.consumeEach { frame ->
-                        if (!frame.frameType.controlFrame) {
-                            assertEquals("Hello", frame.buffer.copy().array().toString(Charsets.UTF_8))
-                            send(frame.copy())
-                            flush()
-                            cancel()
-                        }
-                    }
-                }
-            }
-
-            handleWebSocket("/echo") {
-                setBody(hex("""0x81 0x85 0x37 0xfa 0x21 0x3d 0x7f 0x9f 0x4d 0x51 0x58""".trimHex()))
-            }.let { call ->
-                call.response.awaitWebSocket(Duration.ofSeconds(10))
-
-                val bb = ByteBuffer.wrap(call.response.byteContent!!)
-                assertEquals(11, bb.remaining())
 
-                val parser = FrameParser()
-                parser.frame(bb)
+        val client = createWebSocketsClient()
 
-                assertTrue { parser.bodyReady }
-                assertTrue { parser.mask }
-                val key = parser.maskKey!!
-
-                val collector = SimpleFrameCollector()
-                collector.start(parser.length.toInt(), bb)
-
-                assertFalse { collector.hasRemaining }
+        client.webSocket("/echo") {
+            val frame = incoming.receive()
+            assertEquals(0xc123, frame.data.size)
+        }
 
-                assertEquals("Hello", collector.take(key).copy().array().toString(Charsets.UTF_8))
-            }
+        client.webSocket("/receiveSize") {
+            send("+".repeat(0xcdef))
+            assertEquals("0000cdef", hex(incoming.receive().readBytes()))
         }
     }
 
     @Test
-    fun testSendClose() {
-        withTestApplication {
-            application.install(WebSockets)
+    fun testSendClose() = testApplication {
+        install(WebSockets)
 
-            application.routing {
-                webSocket("/echo") {
-                    incoming.consumeEach { }
-                }
-            }
+        routing {
+            webSocket("/") {}
+        }
 
-            handleWebSocket("/echo") {
-                setBody(hex("""0x88 0x02 0xe8 0x03""".trimHex()))
-            }.let { call ->
-                call.response.awaitWebSocket(Duration.ofSeconds(10))
-                assertEquals("0x88 0x02 0xe8 0x03".trimHex(), hex(call.response.byteContent!!))
-            }
+        var receivedClose = false
+        createWebSocketsClient().webSocketRaw(path = "/") {
+            send(Frame.Close())
+            receivedClose = incoming.receive() is Frame.Close
```

Comment:
I'm not sure about this test. It seems it should check that a client is able to send and receive `Frame.Close`, am I right?